edkrepo/commands/list_repos_command.py

```diff
@@ -183,7 +184,12 @@ def run_command(self, args, config):
                     sources = manifests[project_name].get_repo_sources(combo)
                     for source in sources:
                         if self.get_repo_url(source.remote_url) == repo:
-                            branches.add(source.branch)
+                            if source.branch:

```

Comment:
I made a test Combination that uses commit and tag and branch at the same time.
When the combo is checked out, the priority is supposed to be commit > tag > branch

```
    <Combination name="sandbox-listRepos-test2-commit-tag-branch" description="a combo for testing the list-repos command with commit > tag > branch priority">
      <Source localRoot="edkrepo" remote="EdkrepoRepo" commit="31bb065c7f288437e2823a95606928325d6b2467" tag="edkrepo-v3.1.1" branch="main" />
      <Source localRoot="edkrepo-manifest" remote="EdkrepoManifestRepo" commit="498fc403d3ab9fd96fbd9ac3ca78dd01e5ead466" branch="main" />
    </Combination>
```

The edkrepo list-repos command only lists this Combination by the branch name.

```
edkrepo - [https://github.com/tianocore/edk2-edkrepo]
Branches:
  main
    EdkRepo                                      : *(main)*
    Edkrepo-Sandbox2                             : *(sandbox-main2)*
                                                    (sandbox-listRepos-test2-commit-tag-branch)
edkrepo-manifest - [https://github.com/tianocore/edk2-edkrepo-manifest]
Branches:
  main
    EdkRepoManifest                              : *(main)*
    Edkrepo-Sandbox2                             : *(sandbox-main2)*
                                                    (sandbox-listRepos-test2-commit-tag-branch)
```

Should "edkrepo list-repos" follow the same priority of commit > tag > branch?

Reply:
We should probably show all the commits/tags/branches that are being used, but only put the "*" and green color around the one that is used by default.